Source/PINDiskCache.m

```diff
@@ -854,17 +880,22 @@ - (id)objectForKeyedSubscript:(NSString *)key
     NSDate *now = [[NSDate alloc] init];
     
     [self lock];
-        BOOL isEmpty = (_dates.count == 0 && _sizes.count == 0);
-        BOOL containsKey = [_knownKeys containsObject:key];
+        BOOL containsKey = [_knownKeys containsObject:key] || _diskStateKnown == NO;
     [self unlock];
 
-    if (!key || isEmpty || !containsKey)
+    if (!key || !containsKey)
         return nil;
     
     id <NSCoding> object = nil;
     NSURL *fileURL = [self encodedFileURLForKey:key];
     
     [self lock];
+        if (self->_ttlCache) {
+            // We actually need to know the entire disk state if we're a TTL cache.
+            [self unlock];
+            [self lockUntilDiskStateKnown];
```

Comment:
Why is that? Can't we just return nil for any objectForKey: before initialization is done, including for a TTL cache?

I think I recall there being a default TTL of 30 days. We should consider turning this off by default if it incurs a meaningful performance penalty, which blocking on cache init would be a pretty big cost if the developer hasn't specifically indicated they want TTL behavior.

Reply:
TTL is off by default. It was a feature added by a community member which *enforces* ttl (I.e. it won't return an object even if it has it if it exceeds the TTL). There are separate time limits are probably the default you're think of but they only relate to trimming, not a guarantee something won't be returned. To be frank I'm a bit regretful we merged in this feature, it's not really in the spirit of the framework and adds a lot of complexity.